src/coreclr/jit/lowerxarch.cpp

```diff
@@ -130,6 +130,21 @@ void Lowering::LowerStoreIndir(GenTreeIndir* node)
             return;
         }
     }
+    else if (!node->HasIndex() && node->gtGetOp2()->OperIs(GT_CNS_DBL) &&
+             (node->gtGetOp2()->AsDblCon()->gtDconVal == 0.0))
```

Comment:
Hmm, does this end up optimizing `x = -0.0;` to `mov [x], 0`? Should we check the bit pattern instead?

Reply:
@jakobbotsch good point!

Reply:
Also, maybe it's worth to do it more generally than just 0, i.e. for any 4-byte float immediate?

Reply:
Thanks! changed the code to handle all kind of constants